demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -0,0 +1,644 @@
+ *
+ * The helper functions this demo uses for MQTT operations have internal
+ * loops to process incoming messages. Those are not the focus of this demo
+ * therefore placed in a separate file shadow_demo_helpers.c.
+ */
+int main( int argc,
+          char ** argv )
+{
+    int returnStatus = EXIT_SUCCESS;
+    /* A buffer containing the update document. It has static duration to prevent
+    * it from being placed on the call stack. */
+    static char updateDocument[ SHADOW_REPORTED_JSON_LENGTH + 1 ] = { 0 };
+    int state = 1;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    returnStatus = establishMqttSession( eventCallback );
+
+    if( returnStatus == EXIT_FAILURE )
+    {
+        /* Log error to indicate connection failure. */
+        LogError( ( "Failed to connect to MQTT broker." ) );
+    }
+    else
+    {
+        /* First of all, try to delete any Shadow document in the cloud. */
+        returnStatus = publishToTopic( SHADOW_TOPIC_STRING_DELETE( THING_NAME ),
+                                       SHADOW_TOPIC_LENGTH_DELETE( THING_NAME_LENGTH ),
+                                       updateDocument,
+                                       0U );
+
+        /* Successfully connect to MQTT broker, the next step is
+         * to subscribe shadow topics. */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            returnStatus = subscribeToTopic( SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
+                                             SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            returnStatus = subscribeToTopic( SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ),
+                                             SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            returnStatus = subscribeToTopic( SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ),
+                                             SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
+        }
+        /* This demo uses a constant #THING_NAME known at compile time therefore we can use macros to
+         * assemble shadow topic strings.
+         * If the thing name is known at run time, then we could use the API #Shadow_GetTopicString to
+         * assemble shadow topic strings, here is the example for /update/delta:
+         *
+         * For /update/delta:
+         *
+         * #define SHADOW_TOPIC_MAX_LENGTH  (256U)
+         *
+         * ShadowStatus_t shadowStatus = SHADOW_STATUS_SUCCESS;
+         * char topicBuffer[ SHADOW_TOPIC_MAX_LENGTH ] = { 0 };
+         * uint16_t bufferSize = SHADOW_TOPIC_MAX_LENGTH;
+         * uint16_t outLength = 0;
+         *
+         * shadowStatus = Shadow_GetTopicString( SHADOW_TOPIC_STRING_TYPE_UPDATE_DELTA,
+         *                                       TEST_THING_NAME,
```

Comment:
Better to write variable names `thingName` and `thingNameLength` to emphasize that these are not compile time constants.

Reply:
got it.

Reply:
This [commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1104/commits/49b40ee1665d75b8ed186c326788b6200655e313) should fix the comment.
please let me know if you have other questions.